x/wasm/keeper/ante.go

```diff
@@ -57,14 +57,22 @@ func decodeHeightCounter(bz []byte) (int64, uint32) {
 // LimitSimulationGasDecorator ante decorator to limit gas in simulation calls
 type LimitSimulationGasDecorator struct {
 	gasLimit *sdk.Gas
+	// inputs: simulate, ctx, gas limit, tx
+	gasMeterSetter func(bool, sdk.Context, uint64, sdk.Tx) sdk.Context
```

Comment:
so does this mean before, wasmd simulation wasn't accurate?

Reply:
No it was correct before. But we didn’t use the gas multiplier (NewMultiplierGasMeter) previously and it was set to 1 so it had no effect for simulation

Reply:
Also this doesn’t have to do with wasmd simulation. It’s just that in this wasmd ante, we replace the gas meter with a new one when it’s in simulation mode. That new one doesn’t incorporate the gas multiplier. So this PR just allows us to pass in NewMultiplierGasMeter here